homeassistant/components/geniushub/binary_sensor.py

```diff
@@ -1,52 +1,46 @@
 """Support for Genius Hub binary_sensor devices."""
-from typing import Any, Dict
-
 from homeassistant.components.binary_sensor import BinarySensorDevice
-from homeassistant.util.dt import utc_from_timestamp
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
 
-from . import DOMAIN, GeniusEntity
+from . import CONF_HUB_UID, DOMAIN, GeniusDevice
 
-GH_IS_SWITCH = ["Dual Channel Receiver", "Electric Switch", "Smart Plug"]
+GH_STATE_ATTR = "outputOnOff"
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistantType, config: ConfigType, async_add_entities, discovery_info=None
+):
     """Set up the Genius Hub sensor entities."""
-    client = hass.data[DOMAIN]["client"]
+    if discovery_info is None:
+        return
+
+    hub = hass.data[DOMAIN]["client"]
+    hub_uid = hub.uid if hub.uid else hass.data[DOMAIN][CONF_HUB_UID]
```

Comment:
We should move this to the broker to centralize it, since it's the same check for all platforms.

We should also not store the client specifically in hass.data, but just as an attribute on the broker. We should have a single instance that is saved in hass.data with all central information.

Reply:
Done